db/sql/src/protocol.rs

```diff
@@ -493,13 +514,32 @@ impl HoprDbProtocolOperations for HoprDb {
                     tracing::trace!(pseudonym = %incoming.sender, num_surbs, "stored incoming surbs for pseudonym");
                 }
 
-                Ok(TransportPacketWithChainData::Final {
-                    packet_tag: incoming.packet_tag,
-                    previous_hop: incoming.previous_hop,
-                    plain_text: incoming.plain_text,
-                    ack_key: incoming.ack_key,
-                    no_ack: incoming.no_ack,
-                })
+                // The contained payload represents an Acknowledgement
+                if incoming.ack_key.is_none() {
+                    // TODO: Optimize for more than 1 ACK in the payload
```

Comment:
@coderabbitai Please create an issue to add an optimization to send multiple different acks in a single payload.